### PR TITLE
microws 0.14.5 (new formula)

### PR DESCRIPTION
**Pull request by alebcay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/microws.rb
Comment (ilovezfs) [inline Formula/microws.rb]: We'll need upstream to fix their "make install" target (https://github.com/uNetworking/uWebSockets/blob/master/Makefile#L20-L23) so that it can handle
```
make install PREFIX=/usr/local/Cellar/microws/0.14.4
```
rather than hard coding its prefix.

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/microws.rb
Comment (alebcay) [reply Formula/microws.rb]: Request for upstream to modify the Makefile is [here](https://github.com/uNetworking/uWebSockets/pull/751).

**Comment by alexhultman** [comment]
Shouldn't this be called "uws" instead? brew install uws

**Comment by alebcay** [comment]
I considered "uws" vs "microws" vs "microwebsockets" vs "uwebsockets". I decided on "microws" after reading


> µWS ("microWS") is one of the most lightweight, efficient & scalable WebSocket & HTTP server    implementations available.


Since it's read "microWS" I went with that, figuring that the jump from "uWS" => "µWS". would trip up some people. Just my thought in the moment; others are free to weigh in on how naming conventions apply here, of course.

**Comment by alexhultman** [comment]
Sure, I've also thought about changing the spelling to microws since most americans have no idea of the metric system :wink: :us: Nobody knows that the project is called, not even me

File: Formula/microws.rb
Comment (alebcay) [reply Formula/microws.rb]: Upstream has merged change, but changes in Makefile will not be available until next release, so this formula will have to be on hold until then.

Comment (alexhultman) [reply Formula/microws.rb]: I can make a release if you want to

Comment (alebcay) [reply Formula/microws.rb]: If you think it's a big enough change to warrant it, feel free to. I personally have no rush to get this merged, so it can wait if there haven't been many changes since the last release.

Comment (alexhultman) [reply Formula/microws.rb]: Hahaaa, I've already released it.

**Comment by alebcay** [comment]
Formula now uses `make install PREFIX=...` without any manual installation sorcery.

File: Formula/microws.rb
Comment (DomT4) [inline Formula/microws.rb]: `.opt_include` and `.opt_lib`.

File: Formula/microws.rb
Comment (DomT4) [inline Formula/microws.rb]: Don't need the full path when you're referencing its own keg. Just `#{include}` and `#{lib}`.

File: Formula/microws.rb
Comment (DomT4) [inline Formula/microws.rb]: Should use the `opt.` paths here too.

File: Formula/microws.rb
Comment (DomT4) [inline Formula/microws.rb]: `system "./test"`

File: Formula/microws.rb
Comment (ilovezfs) [inline Formula/microws.rb]: `-std=c++14` not `--std=c++14`

Do we need to add `needs :cxx14` to the formula too?

File: Formula/microws.rb
Comment (ilovezfs) [inline Formula/microws.rb]: is the separate `make` step actually required?

File: Formula/microws.rb
Comment (ilovezfs) [inline Formula/microws.rb]: Let's wrap these args in such a way as to avoid going passed the 80th column

File: Formula/microws.rb
Comment (ilovezfs) [inline Formula/microws.rb]: is this port actually used by the test? if so we shouldn't hard code it.

**Review by ilovezfs — CHANGES_REQUESTED** [review]

**Comment by ilovezfs** [comment]
Regarding naming … has this been packaged in any other package managers yet?

**Comment by alexhultman** [comment]
It has been added to various package managers under various names by various people, things has not really been coordinated under one single name but uws, uwebsockets, uWS are most common I guess.

The only package manager I publish to is NPM under the name uws: https://www.npmjs.com/package/uws

If uws is free then that would be a good name IMO.

File: Formula/microws.rb
Comment (stephengroat) [inline Formula/microws.rb]: maybe include a `head` with `head "https://github.com/uNetworking/uWebSockets.git"`?

Comment (DomT4) [reply Formula/microws.rb]: Sometimes it's worth holding off to see what kind of demand there is for a `head` spec. They can add quite a bit of support burden sometimes; people often seem to expect Homebrew to provide support when `HEAD` builds fail but Homebrew practically speaking can't provide support for a moving target.

I defer to ILZ for an opinion here though.

Comment (stephengroat) [reply Formula/microws.rb]: i only think it makes sense if the formula is using `github.com/user/project/archive/version`

since that's just a tar of the source, it shouldn't be a moving target (until someone makes a change that won't work for the next release anyway)

Comment (DomT4) [reply Formula/microws.rb]: Yeah, I meant that `HEAD` is a moving target. Something that is broken today may be fixed tomorrow, and indeed, vice versa.

Comment (stephengroat) [reply Formula/microws.rb]: gotcha, excellent point.

i was referring to `HEAD` changes in build procedures (specifically)

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/microws.rb
Comment (ilovezfs) [reply Formula/microws.rb]: Yeah `head` is not something we want to routinely add except when someone comes asking for it for their own use.

**Comment by ilovezfs** [comment]
@alebcay any updates here?

**Comment by alebcay** [comment]
Was just out of town until yesterday. I'm thinking to change the name to `uws` for consistency with `npm`.

**Comment by ilovezfs** [comment]
and archlinux is calling it uwebsockets https://aur.archlinux.org/packages/uwebsockets/

**Comment by ilovezfs** [comment]
Let's call it uws and make uwebsockets an alias

File: Formula/microws.rb
Comment (alebcay) [reply Formula/microws.rb]: No, it doesn't actually use it. The test simply tries to bind to a port; any port should be fine.

File: Formula/microws.rb
Comment (alebcay) [reply Formula/microws.rb]: Yes. Otherwise `make install` complains about files being missing (i.e. `make install` only installs and does not build/compile).

File: Formula/microws.rb
Comment (ilovezfs) [reply Formula/microws.rb]: Right but does it try to bind 3000?

File: Formula/microws.rb
Comment (alexhultman) [reply Formula/microws.rb]: It is C++11

**Comment by alexhultman** [comment]
So are you going to add this or not I don't get the point of being very active and then just letting it auto-close?

**Comment by ilovezfs** [comment]
@alexhultman it wasn't renamed as requested and I'm not in the habit of regularly finishing PRs for people.

**Comment by alexhultman** [comment]
@ilovezfs I get that but according to your rules I'm not allowed to fix it.

**Comment by ilovezfs** [comment]
@alexhultman not sure what rule that is. If alebcay doesn't want to fix it himself, he can give you commit access to his fork.

**Comment by alexhultman** [comment]
I'm pretty sure I read somewhere that Homebrew does not want authors to add their own projects.

**Comment by ilovezfs** [comment]
@alebcay https://github.com/Homebrew/brew/blob/master/docs/Acceptable-Formulae.md#niche-or-self-submitted-stuff

>We frown on authors submitting their own work unless it is very popular.

Given your project has 352 watchers, 8,517 stares, and 634 forks, we'd not even be bending that rule :)

**Comment by alexhultman** [comment]
Ah. Well in that case I could probably look at it some time.

**Comment by alebcay** [comment]
> @alexhultman not sure what rule that is. If alebcay doesn't want to fix it himself, he can give you commit access to his fork.

Done.

**Pull request closed without merging**
